pkg/operator/status.go

```diff
@@ -148,38 +148,33 @@ func (optr *Operator) syncStatus(co *osconfigv1.ClusterOperator, conds []osconfi
 }
 
 func (optr *Operator) getOrCreateClusterOperator() (*osconfigv1.ClusterOperator, error) {
+	var co *osconfigv1.ClusterOperator
 	co, err := optr.osClient.ConfigV1().ClusterOperators().Get(clusterOperatorName, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		// to report the status of all the managed components.
-		// TODO we will report the version of the operands (so our machine api implementation version)
-		// NOTE: related objects lets openshift/must-gather collect diagnostic content
 		co = &osconfigv1.ClusterOperator{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: clusterOperatorName,
 			},
-			Status: osconfigv1.ClusterOperatorStatus{
-				Versions: optr.operandVersions,
-				RelatedObjects: []osconfigv1.ObjectReference{
-					{
-						Group:    "",
-						Resource: "namespaces",
-						Name:     optr.namespace,
-					},
-				},
-			},
 		}
-
-		glog.Infof("%s clusterOperator status does not exist, creating %v", clusterOperatorName, co)
-		co, err := optr.osClient.ConfigV1().ClusterOperators().Create(co)
+		glog.Infof("clusterOperator %q does not exist, creating a new one: %v", clusterOperatorName, co)
+		co, err = optr.osClient.ConfigV1().ClusterOperators().Create(co)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to create new clusterOperator: %v", err)
 		}
-		return co, nil
 	}
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get clusterOperator %q: %v", clusterOperatorName, err)
+	}
+
+	// Related objects lets openshift/must-gather collect diagnostic content
+	co.Status.RelatedObjects = []osconfigv1.ObjectReference{
+		{
+			Group:    "",
+			Resource: "namespaces",
+			Name:     optr.namespace,
+		},
 	}
-	return co, nil
+	return optr.osClient.ConfigV1().ClusterOperators().UpdateStatus(co)
```

Comment:
This *always* calls `UpdateStatus()`? That seems like a weird side effect of retrieving the ClusterOperator object. I could see this spamming updates in the future if it's not already.